packages/remix-dev/compiler/analysis.ts

```diff
@@ -0,0 +1,20 @@
+import fs from "node:fs";
+import path from "node:path";
+import type { Metafile } from "esbuild";
+
+import type { Context } from "./context";
+
+export let writeMetafile = (
+  ctx: Context,
+  filename: string,
+  metafile: Metafile
+) => {
+  let buildDir = path.dirname(ctx.config.serverBuildPath);
+  if (!fs.existsSync(buildDir)) {
+    fs.mkdirSync(buildDir, { recursive: true });
+  }
+  fs.writeFileSync(
+    path.join(buildDir, filename),
+    JSON.stringify(metafile, undefined, 2)
```

Comment:
Do we care if the file is pretty?

Reply:
probably not actually. I can remove the `, undefined, 2` part